### DISCUSSION TITLE
Availability of Python of CLI

### DISCUSSION THREAD

**Discussion by SchrodingersParakeet** [bonsai-rx/bonsai#1201] [Q&A]
Does Bonsai have a CLI or Python interface? I saw a a mention of the CLI guide on google groups but the link seems to be dead.

I’d really like to do some simple things like start or stop a workflow or get real-time data into Python without using the OSC broadcast.

**Comment by bruno-f-cruz** [comment]
We have recently moved it to the docs repo:
https://bonsai-rx.org/docs/articles/cli.html

**Reply by SchrodingersParakeet** [reply]
Thanks! But can you stop a workflow from the CLI? 

I’m relying on TTLs controlled by the pylon capture module to track position of my rats and I need to stop Bonsai before open ephys to ensure I know when the last title was sent.

**Comment by SchrodingersParakeet** [comment]
Along the lines of #1230 i am curious to see if it’s possible to stop bonsai from the command line you can definitely start it, but simply killing the process (e.g. to pause between trials) corrupts the video

is there any command line or SDK we can use? Or a script that can be run from power shell to gracefully stop things?

Thanks,
Chris
